File: Tests/AndroidSDKTestApp/AndroidSDKTestApp/Delegates/OPIdentityLookupDelegateImplementation.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Android.Util;
using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Com.Openpeer.Delegates;
using Com.Openpeer.Javaapi;
namespace AndroidSDKTestApp
{
	public class OPIdentityLookupDelegateImplementation:OPIdentityLookupDelegate
	{
		public override void OnIdentityLookupCompleted (OPIdentityLookup lookup)
		{
			Log.Debug("output", "Identity lookup test PASSED");
			LoginManager.onIdentityLookupCompleted(lookup);
		}
	}
}

File: openpeer-android-sdk/src/com/openpeer/sdk/app/OPIdentityLookupDelegateImplementation.java
package com.openpeer.sdk.app;

import java.util.Hashtable;
import java.util.List;

import android.util.Log;

import com.openpeer.javaapi.OPIdentity;
import com.openpeer.javaapi.OPIdentityContact;
import com.openpeer.javaapi.OPIdentityLookup;
import com.openpeer.javaapi.OPIdentityLookupDelegate;

public class OPIdentityLookupDelegateImplementation extends OPIdentityLookupDelegate {

	static Hashtable<String, OPIdentityLookupDelegateImplementation> instances = new Hashtable<String, OPIdentityLookupDelegateImplementation>();

	public static OPIdentityLookupDelegateImplementation getInstance(OPIdentity identity) {
		String url = identity.getIdentityURI();
		OPIdentityLookupDelegateImplementation instance = instances.get(url);
		if (instance == null) {
			instance = new OPIdentityLookupDelegateImplementation();
			instance.mIdentity = identity;
		}
		instances.put(url, instance);
		return instance;
	}

	private OPIdentity mIdentity;

	private OPIdentityLookupDelegateImplementation() {
	}

	public OPIdentityLookupDelegateImplementation(OPIdentity identity) {
		mIdentity = identity;
	}

	@Override
	public void onIdentityLookupCompleted(OPIdentityLookup lookup) {
		String url = mIdentity.getIdentityURI();		Log.d("OPIdentityLookupDelegateImplementation", "onIdentityLookupCompleted " + lookup);

		OPDataManager.getInstance().onIdentityLookupCompleted(url,lookup);
		instances.remove(url);

	}

}
